src/Theme.php

```diff
@@ -0,0 +1,205 @@
+    {
+        static $result = null;
+        if ($result === null) {
+            $result = [
+                new self('aerialgreen', _x('theme', 'Aerial Green'), false, false),
+                new self('auror', _x('theme', 'Auror'), false, false),
+                new self('auror_dark', _x('theme', 'Dark Auror'), true, false),
+                new self('automn', _x('theme', 'Autumn'), false, false),
+                new self('classic', _x('theme', 'Classic'), false, false),
+                new self('clockworkorange', _x('theme', 'Clockwork Orange'), false, false),
+                new self('dark', _x('theme', 'Dark'), false, false),
+                new self('darker', _x('theme', 'Darker'), true, false),
+                new self('flood', _x('theme', 'Flood'), false, false),
+                new self('greenflat', _x('theme', 'Green Flat'), false, false),
+                new self('hipster', _x('theme', 'Hipster'), false, false),
+                new self('icecream', _x('theme', 'Ice Cream'), false, false),
+                new self('lightblue', _x('theme', 'Light Blue'), false, false),
+                new self('midnight', _x('theme', 'Midnight'), true, false),
+                new self('premiumred', _x('theme', 'Premium Red'), false, false),
+                new self('purplehaze', _x('theme', 'Purple Haze'), false, false),
+                new self('teclib', _x('theme', 'Teclib'), false, false),
+                new self('vintage', _x('theme', 'Vintage'), false, false),
+            ];
+        }
+
+        return $result;
+    }
+
+    /**
+     * Get all themes present in the "palettes" directory that aren't core themes
+     * @return Theme[]
+     */
+    public static function getCustomThemes(): array
+    {
+        static $custom_themes = null;
+
+        if ($custom_themes === null) {
+            $custom_themes = [];
+
+            $file_matches = [];
+            // Cannot use GLOB_BRACE on some platforms (like the docker environment used for tests)
+            $patterns = [
+                '*.css',
+                '*.scss'
+            ];
+            foreach ($patterns as $pattern) {
+                foreach (glob(self::CORE_THEME_ROOT . $pattern) as $file) {
+                    $file_name = pathinfo($file, PATHINFO_FILENAME);
+                    if (str_starts_with($file_name, '_')) {
+                        continue;
+                    }
+                    $file_matches[$file_name] = $file;
+                }
+            }
+
+            $core_themes = self::getCoreThemes();
+            $core_keys = [];
+            foreach ($core_themes as $core_theme) {
+                $core_keys[] = $core_theme->getKey();
+            }
+
+            foreach ($file_matches as $file_name => $file) {
+                if (!in_array($file_name, $core_keys, true)) {
+                    // Guess dark mode based on the filename ("dark" with - or _ before or after)
+                    $pattern = '/dark[-_]|[-_]dark/i';
```

Comment:
Using `\b` would handle more cases.
```suggestion
                    $pattern = '/\bdark\b/i';
```